### PR TITLE
Project Update to V4.0.0

### PR DESCRIPTION
**Pull request by InvictusRMC**
## Introduction
This PR merges the latest release of the base repository with ours. This is necessary as this fixes some issues with the Android camera roll: some read in images had epoch timestamps of 0. As a result, this lead to images missing when creating an activity.

## The Changes:
**Note:** Most changes introduced in this PR are through commits from the base repository, the important changes I have made (due to merge conflicts) reside in ```android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java```. As such, I kindly ask you to thoroughly review this file. Also @FrikkieSnyman I note that this repo is a fork of your fork, is it correct for me to make the PR here?

### CODE REVIEW

File: android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java [hunk 1]
```diff
@@ -311,25 +366,39 @@ protected void doInBackgroundGuarded(Void... params) {
...
+
         Cursor media = resolver.query(
-            MediaStore.Files.getContentUri("external"),
+            MediaStore.Files.getContentUri("external").buildUpon().encodedQuery(limit).build(),
             PROJECTION,
             selection.toString(),
             selectionArgs.toArray(new String[selectionArgs.size()]),
-            Images.Media.DATE_TAKEN + " DESC, " + Images.Media.DATE_MODIFIED + " DESC LIMIT " +
-                (mFirst + 1)); // set LIMIT to first + 1 so that we know how to populate page_info
+            Images.Media.DATE_ADDED + " DESC, " + Images.Media.DATE_MODIFIED + " DESC");
```
[the remark this answers is not in the document]
Comment (InvictusRMC) [reply android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:395]: Perhaps make the behaviour different per Android version? Right now taking a picture using the Photo button in the app is broken due to this for Android 10.

[the remark this answers is not in the document]
Comment (bryanvanwijk) [reply android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:395]: Yes, I think we should change this back to `DATE_TAKEN` and if `DATE_TAKEN = 0` only then use `DATE_ADDED`

[the remark this answers is not in the document]
Comment (FrikkieSnyman) [reply android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:395]: Yeah would be ideal to solve it via SQL lite query. But might actually be worth running a different query for >= Android 10 than for < Android 10.

**Comment by InvictusRMC** [comment]
Closing this PR and continuing a temporary fix in #6.

File: android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (InvictusRMC) [reply android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:395]: I came up with a temporary fix in #6.

**Pull request closed without merging**